src/serverenvironment.cpp

```diff
@@ -1070,6 +1070,74 @@ bool ServerEnvironment::swapNode(v3s16 p, const MapNode &n)
 	return true;
 }
 
+u8 ServerEnvironment::findSunlight(v3s16 pos)
+{
+	// Directions for neighbouring nodes with specified order
+	static const v3s16 dirs[] = {
+		v3s16(-1, 0, 0), v3s16(1, 0, 0), v3s16(0, 0, -1), v3s16(0, 0, 1),
+		v3s16(0, -1, 0), v3s16(0, 1, 0)
+	};
+
+	const NodeDefManager *ndef = m_server->ndef();
+	// found_light remembers the highest known sunlight value at pos
+	u8 found_light = 0;
+	struct stack_entry {
+		v3s16 pos;
+		s16 dist;
```

Comment:
Why isn't it `s8` like everywhere else?

Reply:
I used the 16 bit value so that stack_entry takes 64 bit (3*16+16) to ensure that it is aligned in memory. (I don't know if it really makes a difference)